design/xxx-design-crd-v1beta1.md

```diff
@@ -0,0 +1,198 @@
+* Provider: Is a **source** for secrets. The Provider is external to ESO. It can be a hosted service like Alibaba Cloud SecretsManager, AWS SystemsManager, Azure KeyVault etc
+* SecretStore `ST`: A Custom Resource to authenticate and configure the connection between the ESO instance and the Provider
+* ExternalSecret `ES`: A Custom Resource that declares which secrets should be synced
+* Frontend: A **sink** for the synced secrets, usually a `Secret` resource
+* Secret: Credentials that act as a key to sensitive information
+
+### User Definitions
+* `operator :=` I manage one or multiple `ESO` instances
+* `user :=` I only create `ES`, ESO is managed by someone else
+
+### User Stories
+From that we can derive the following requirements or user stories:
+1. As a ESO operator I want to get all the secrets of a given path from a given provider, if the provider supports it
+2. As a ESO operator I want to handle templating like it is a natural language, not needing to worry about how it is actually implemented.
+
+## Proposal
+
+### External Secret
+
+```yaml
+#only changed fields are commented out.
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: "hello-world"
+  labels:
+    acme.org/owned-by: "q-team"
+  annotations:
+    acme.org/sha: 1234
+
+spec:
+  secretStoreRef:
+    name: secret-store-name
+    kind: SecretStore
+  refreshInterval: "1h"
+  target:
+    name: my-secret
+    creationPolicy: 'Merge'
+    template:
+      engineVersion: v2 #Defaults to v2 in v1beta1
+      type: kubernetes.io/dockerconfigjson 
+      metadata:
+        annotations: {}
+        labels: {}
+      data:
+        config.yml: |
+          endpoints:
+          - https://{{ .data.user }}:{{ .data.password }}@api.exmaple.com
+
+      templateFrom:
+      - configMap:
+          name: alertmanager
+          items:
+          - key: alertmanager.yaml
+  data:
+    - secretKey: secret-key-to-be-managed
+      remoteRef:
+        key: provider-key
+        version: provider-key-version
+        property: provider-key-property
+  dataFrom:
+  - extract: #extract all the keys from one given secret
+      key: provider-key
+      version: provider-key-version
+      property: provider-key-property
+  - find:
+      name:  #extract all keys from all secrets that match a pattern
+        regexp: .*pattern.*
+      tags:  #extract all keys from all secrets that have the following labels/tags
+        provider-label: provider-value
+status:
+  refreshTime: "2019-08-12T12:33:02Z"
+  conditions:
+  - type: Ready
+    status: "True"
+    reason: "SecretSynced"
+    message: "Secret was synced"
+    lastTransitionTime: "2019-08-12T12:33:02Z"
+```
+
+#### Behavior
+
+ExternalSecrets now will have a different structure for `dataFrom`, which will allow fetching several provider secrets with only one ExternalSecret definition. It should be possible to fetch secrets based on regular expressions or by a label/tag selector.
+
```

Comment:
```suggestion
If the user desires to rename the secret keys (e.g. because the key name is not a valid secret key name `/foo/bar`) he/she should use `template` functions to produce a mapping. 
```